### PR TITLE
Homework 1.2

### PR DESCRIPTION
**Pull request by Marimeiko**

### CODE REVIEW

File: module1/homework/calculate/calculate.hpp [hunk 1]
```diff
@@ -1,7 +1,30 @@
...
+    else if (command == "divide") {
+        if (command==0){
+            std::cout << "Division by 0";
+        }
+        result = first / second;
+        std::cout << result;
+    }
+    else {
+        std::cout << "Invalid data";
+    }
```
Comment (piezol) [inline module1/homework/calculate/calculate.hpp:28]: I've put my comments on the wrong pull request so once again here:
Calculate function should not have any side effects (ie we should not use std::cout).

File: module1/homework/fibonacci/fibonacci.hpp [hunk 2]
```diff
@@ -1,11 +1,32 @@
...
+
+    int first = 1;
+    int second = 1;
+    int result;
+
+    if (sequence <= 3){
+        return 1;
+    }
+    
+    for (int i = 3; i <= sequence; i++) {
```
Comment (piezol) [inline module1/homework/fibonacci/fibonacci.hpp:14]: Fibonnaci (3) is not equal to 1.

File: module1/homework/calculate/calculate.hpp [hunk 3]
```diff
@@ -1,7 +1,30 @@
 #pragma once
 #include <string>
 
+int result;
```
Comment (kyrtaq) [inline module1/homework/calculate/calculate.hpp:4]: put it inside the calculate function, it`s bad practice to make global variables

File: module1/homework/fibonacci/fibonacci.hpp [hunk 4]
```diff
@@ -1,11 +1,32 @@
...
-    // TODO: Your implementation goes here
-    return 0;
+
+    int first = 1;
+    int second = 1;
+    int result;
+
+    if (sequence <= 3){
+        return 1;
+    }
```
Comment (piezol) [inline module1/homework/fibonacci/fibonacci.hpp:12]: You're not handling fib(0) at all.

File: module1/homework/calculate/calculate.hpp [hunk 5]
```diff
@@ -1,7 +1,30 @@
...
+    else if (command == "add") {
+        result = first + second;
+        std::cout << result;
+    }
+    else if (command == "subtract") {
+        result = first - second;
+        std::cout << result;
+    }
+    else if (command == "divide") {
+        if (command==0){
```
Comment (kyrtaq) [inline module1/homework/calculate/calculate.hpp:20]: you can`t compare std::string with int, did you mean to compare it with the variable called second?

File: module1/homework/calculate/calculate.hpp [hunk 6]
```diff
@@ -1,7 +1,30 @@
 #pragma once
 #include <string>
 
+int result;
+
 std::string calculate(const std::string& command, int first, int second) {
```
Comment (kyrtaq) [inline module1/homework/calculate/calculate.hpp:6]: you should return the result as a std:;string, I suggest you to use std::to_string

File: module1/homework/fibonacci/fibonacci.hpp [hunk 7]
```diff
@@ -1,11 +1,32 @@
...
-    // TODO: Your implementation goes here
-    return 0;
+
+    if(sequence == 0) {
+        return 0;
 }
+    else if (sequence < 3) {
+      return 1;
+}
+ return fibonacci_recursive(sequence-2)+fibonacci_recursive(sequence-1);
```
Comment (kuchichan) [inline module1/homework/fibonacci/fibonacci.hpp:30]: Spaces around operator. Please consider to use some autoformatter - based on Q&A lesson #1 :)

Comment (DamiXioR) [reply module1/homework/fibonacci/fibonacci.hpp:30]: it's better to check if sequence is lesser than 2 and return sequence

File: module1/homework/fibonacci/fibonacci.hpp [hunk 8]
```diff
@@ -1,11 +1,32 @@
...
 
 int fibonacci_recursive(int sequence) {
-    // TODO: Your implementation goes here
-    return 0;
+
+    if(sequence == 0) {
+        return 0;
 }
+    else if (sequence < 3) {
+      return 1;
```
Comment (Marcinati) [inline module1/homework/fibonacci/fibonacci.hpp:28]: Or just:
if (sequence < 2) {
return sequence;
}

and other return will cover the rest (from 2).

File: module1/homework/fibonacci/fibonacci.hpp (hunk 4 above)
Comment (WBigaj) [reply module1/homework/fibonacci/fibonacci.hpp:12]: And negative values.

File: module1/homework/fibonacci/fibonacci.hpp (hunk 8 above)
Comment (WBigaj) [reply module1/homework/fibonacci/fibonacci.hpp:28]: Sequence < 2 is bad when int is negative.

**Comment by ziobron** [comment]
Let me close this PR. If you wish to have some points please fix it, deliver and re-open this Pull Request.

**Pull request closed without merging**